Annotate NetworkStorageManager endpoints with feature flags

darryl-apple opened the pull request:
#### ade42ab52ee0d7a8692d25228090a6716ac9b626
<pre>
Annotate NetworkStorageManager endpoints with feature flags
<a href="https://bugs.webkit.org/show_bug.cgi?id=281329">https://bugs.webkit.org/show_bug.cgi?id=281329</a>
<a href="https://rdar.apple.com/137767816">rdar://137767816</a>

Reviewed by NOBODY (OOPS!).

Annotate NetworkStorageManager endpoints with feature flags

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.messages.in:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::processForConnection):
* Source/WebKit/UIProcess/WebProcessProxy.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ade42ab52ee0d7a8692d25228090a6716ac9b626

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82042 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1569 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35999 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86599 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33074 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1604 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9395 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63955 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21703 "Found 49 new test failures: http/tests/inspector/dom/cross-domain-inspected-node-access.html http/tests/inspector/network/beacon-type.html http/tests/inspector/network/copy-as-curl.html http/tests/inspector/network/copy-as-fetch.html http/tests/inspector/network/fetch-network-data.html http/tests/inspector/network/fetch-response-body-304.html http/tests/inspector/network/getSerializedCertificate.html http/tests/inspector/network/har/har-basic.html http/tests/inspector/network/har/har-import.html http/tests/inspector/network/intercept-request-fragment.html ... (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85112 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1190 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74645 "Found 2 new API test failures: TestWebKitAPI.WKWebView.LocalStorageProcessCrashes, TestWebKitAPI.WKWebsiteDataStoreConfiguration.TotalQuotaRatio (failure)") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44239 "Found 12 new API test failures: /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/cache, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/itp, /WPE/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/navigation-policy, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/handle-corrupted-local-storage, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/configuration, /WPE/TestSSL:/webkit/WebKitWebView/web-socket-tls-errors, /WPE/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/autoplay-policy, /WPE/TestWebKitWebView:/webkit/WebKitWebView/is-playing-audio, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/databases, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/storage ... (failure)") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1092 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-anchor-position/anchor-scroll-position-try-006.html imported/w3c/web-platform-tests/html/cross-origin-opener-policy/iframe-popup-same-origin-to-unsafe-none.https.html?5-6 (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28822 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31494 "Built successfully") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/75023 "Found 1 new JSC stress test failure: stress/scoped-arguments-table-should-be-tolerant-for-oom.js.no-cjit-collect-continuously (failure)") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72408 "Found 1 new API test failure: TestWebKitAPI.WKWebsiteDataStoreConfiguration.TotalQuotaRatio (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29434 "Found 8 new test failures: http/tests/cookies/same-site/fetch-after-top-level-navigation-from-cross-origin-page.html http/tests/referrer-policy-anchor/same-origin/cross-origin-http-http.html http/tests/resourceLoadStatistics/cookie-deletion.html http/tests/resourceLoadStatistics/no-third-party-cookie-blocking-when-itp-is-off.html http/tests/security/contentSecurityPolicy/form-action-src-redirect-allowed2.html http/tests/storageAccess/request-and-grant-access-then-detach-should-not-have-access.https.html imported/w3c/web-platform-tests/html/cross-origin-opener-policy/iframe-popup-same-origin-to-unsafe-none.https.html?5-6 imported/w3c/web-platform-tests/html/cross-origin-opener-policy/iframe-popup-same-origin-to-unsafe-none.https.html?9-last (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88032 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/81096 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9284 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7368 "Found 30 new test failures: fast/selectors/selection-window-inactive-stroke-color.html http/tests/cache/partitioned-cache-iframe.html http/tests/cache/partitioned-cache.html http/tests/cookies/same-site/fetch-after-top-level-navigation-initiated-from-iframe-in-cross-origin-page.html http/tests/inspector/target/target-events-for-provisional-page.html http/tests/misc/form-submit-file-cross-site.html http/tests/navigation/sec-fetch-site-header.html http/tests/privateClickMeasurement/database-disabled-in-ephemeral-session.html http/tests/privateClickMeasurement/private-click-measurement-with-source-nonce-wrong-content-type.html http/tests/privateClickMeasurement/second-attribution-converted-with-lower-priority.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72323 "Failure limit exceed. At least found 266 new test failures: http/tests/inspector/dom/cross-domain-inspected-node-access.html http/tests/inspector/dom/didFireEvent.html http/tests/inspector/dom/disconnect-dom-tree-after-main-frame-navigation.html http/tests/inspector/gatherWebInspectorRTCLogs.html http/tests/inspector/network/beacon-type.html http/tests/inspector/network/contentextensions/blocked-websocket-crash.html http/tests/inspector/network/copy-as-curl.html http/tests/inspector/network/copy-as-fetch.html http/tests/inspector/network/eventsource-type.html http/tests/inspector/network/fetch-network-data.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9469 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70464 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71545 "Found 100 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitSecurityManager/file-xhr, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/test-page-list, /WebKitGTK/TestWebKitUserContentManager:/webkit/WebKitUserContentManager/injected-script, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/notification-initial-permission-allowed, /WebKitGTK/TestSSL:/webkit/WebKitWebView/load-failed-with-tls-errors, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/enable-html5-database, /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/get-cookies, /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/query-permission-requests, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/usermedia-enumeratedevices-permission-check ... (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15669 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14581 "Found 1 new test failure: storage/indexeddb/modern/opendatabase-after-storage-crash.html (failure)") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/689 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9235 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14771 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/103508 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9075 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25120 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12601 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10883 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -196,6 +196,15 @@ RefPtr<WebProcessProxy> WebProcessProxy::processForIdentifier(ProcessIdentifier
     return allProcessMap().get(identifier);
 }
 
+RefPtr<WebProcessProxy> WebProcessProxy::processForConnection(const IPC::Connection& connection)
```
This change should not be needed for this patch

darryl-apple committed:
Annotate NetworkStorageManager endpoints with feature flags
https://bugs.webkit.org/show_bug.cgi?id=281329
rdar://137767816

Reviewed by NOBODY (OOPS!).

Annotate NetworkStorageManager endpoints with feature flags

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp:
(WebKit::NetworkStorageManager::sharedPreferencesForWebProcess const):
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.h:
* Source/WebKit/NetworkProcess/storage/NetworkStorageManager.messages.in:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::processForConnection):
* Source/WebKit/UIProcess/WebProcessProxy.h:
